n3fit/src/evolven3fit_new/utils.py

```diff
@@ -0,0 +1,179 @@
+            raise ValueError("The q2 requested is not the fitting scale of this pdf")
+        return self.funcs[list(PIDS_DICT.values()).index(PIDS_DICT[pid])](x)
+
+    def hasFlavor(self, pid):
+        """Check if the requested pid is in the PDF."""
+        return pid in PIDS_DICT.keys()
+
+
+def read_runcard(usr_path):
+    """Read the runcard and return the relevant information for evolven3fit_new"""
+    return yaml.safe_load((usr_path / "filter.yml").read_text())
+
+
+def get_theoryID_from_runcard(usr_path):
+    """Return the theoryID from the runcard"""
+    # read the runcard
+    my_runcard = read_runcard(usr_path)
+    return my_runcard["theory"]["theoryid"]
+
+
+def generate_q2grid(Q0, Qfin, Q_points, match_dict):
+    """Generate the q2grid used in the final evolved pdfs or use the default grid if Qfin or Q_points is
+    not provided.
+    """
+    if Qfin is None:
+        if Q_points is None:
+            return (
+                np.array(
+                    [
+                        1.6500000e00,
+                        1.7874388e00,
+                        1.9429053e00,
+                        2.1193749e00,
+                        2.3204100e00,
+                        2.5502944e00,
+                        2.8142025e00,
+                        3.1184122e00,
+                        3.4705775e00,
+                        3.8800751e00,
+                        4.3584516e00,
+                        4.9200000e00,
+                        4.9200000e00,
+                        5.5493622e00,
+                        6.2897452e00,
+                        7.1650687e00,
+                        8.2052867e00,
+                        9.4481248e00,
+                        1.0941378e01,
+                        1.2745972e01,
+                        1.4940062e01,
+                        1.7624572e01,
+                        2.0930715e01,
+                        2.5030298e01,
+                        3.0149928e01,
+                        3.6590777e01,
+                        4.4756282e01,
+                        5.5191298e01,
+                        6.8637940e01,
+                        8.6115921e01,
+                        1.0903923e02,
+                        1.3938725e02,
+                        1.7995815e02,
+                        2.3474820e02,
+                        3.0952544e02,
+                        4.1270732e02,
+                        5.5671861e02,
+                        7.6011795e02,
+                        1.0509694e03,
+                        1.4722574e03,
+                        2.0906996e03,
+                        3.0112909e03,
+                        4.4016501e03,
+                        6.5333918e03,
+                        9.8535186e03,
+                        1.5109614e04,
+                        2.3573066e04,
+                        3.7444017e04,
+                        6.0599320e04,
+                        1.0000000e05,
+                    ]
```

Comment:
Sure I can call it muf2. The grid is actually our default grid. We looked inside apfel to understand how it is generated and it turns out that it is a bit more complicate than just a `np.geomspace`. Therefore we decided to avoid reproducing it. So, yes, I am putting it top level.